src/arrayMethodSort.js

```diff
@@ -4,9 +4,28 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compareFunc =
+      compareFunction ||
+      function (a, b) {
+        if (`${a}` > `${b}`) {
+          return 1;
+        } else if (`${a}` < `${b}`) {
+          return -1;
+        }
+
+        return 0;
+      };
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compareFunc(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm you implemented here is correct, but it's not the most efficient sorting algorithm. It has a time complexity of O(n^2), which means it can be very slow for large arrays. If performance is a concern, you might want to consider using a more efficient sorting algorithm, like quicksort or mergesort. However, for the purpose of this task, the bubble sort algorithm is acceptable.